src/System.IO.FileSystem.Watcher.Polling/System/IO/PollingWatcher.cs

```diff
@@ -84,43 +88,20 @@ private FileChangeList ComputeChangesAndUpdateState()
             _version++;
             var changes = new FileChangeList();
 
-            WIN32_FIND_DATAW fileData = new WIN32_FIND_DATAW();
-            unsafe
+            if (Directory.Exists(_directory))
             {
-                WIN32_FIND_DATAW* pFileData = &fileData;
-
-                for(int index=0; index < _directories.Count; index++) {
-                    var directory = _directories[index];
-
-                    var handle = DllImports.FindFirstFileExW(directory, FINDEX_INFO_LEVELS.FindExInfoBasic, pFileData, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, DllImports.FIND_FIRST_EX_LARGE_FETCH);
-                    if (handle == DllImports.INVALID_HANDLE_VALUE) { // directory got deleted 
-                        if (Tracing.Switch.ShouldTrace(TraceEventType.Information))
-                        {
-                            Tracing.TraceEvent(TraceEventType.Warning, 2, "Directory could not be opened {0}", directory);
-                        }
-                        _directories.Remove(directory);
-                        continue;
-                    }
-
-                    try
-                    {
-                        do
-                        {
-                            if (IsSpecial(fileData.cFileName)) continue;
-                            if (!IsWatched(fileData.cFileName)) continue;
-                            UpdateState(directory, ref changes, ref fileData, fileData.cFileName);
-                        }
-                        while (DllImports.FindNextFileW(handle, pFileData));
-                    }
-                    finally
-                    {
-                        DllImports.FindClose(handle);
-                    }
+                var files = Directory.EnumerateFiles(_directory, "*", _includeSubdirectories ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
```

Comment:
Making this change would significantly impact the performance of the watcher as it will now allocate full path strings for every file when there were no allocations before.

Reply:
Dumb question: where do I find `FileSystemEnumerable<T>`?

Reply:
In the `System.IO.FileSystem` package.

Reply:
It is in the System.IO.Enumeration namespace: https://apisof.net/catalog/System.IO.Enumeration.FileSystemEnumerable%3CTResult%3E

Reply:
I had assumed that was the package it would be in, but the [last NuGet package](https://www.nuget.org/packages/System.IO.FileSystem/4.3.0) was in 2016 and the [last MyGet package](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.IO.FileSystem/4.4.0-beta-24913-02) was in the beginning of 2017. So where do I find this updated package?

Reply:
@KrzysztofCwalina, how do we consume the latest bits in a project in CoreFXLab?

Reply:
If I create a brand new Core console app (and change the target to 2.1), I still can't find the updated code. I would expect it to be in NuGet marked "Preview", but like I said that code hasn't been updated in a while. So while my issue is certainly in CoreFXLab, the issue extends to "normal" development.

Reply:
> how do we consume the latest bits in a project in CoreFXLab?

These are [netstandard libraries](https://github.com/dotnet/corefxlab/blob/master/src/System.IO.FileSystem.Watcher.Polling/System.IO.FileSystem.Watcher.Polling.csproj#L5), so what exactly do you want to consume in corefxlab?

> If I create a brand new Core console app (and change the target to 2.1), I still can't find the updated code.

Aren't System.IO.FileSystem/Enumeration part of [Microsoft.NetCoreApp](https://www.nuget.org/packages/Microsoft.NETCore.App/2.1.0-preview2-26406-04) package that comes as part of 2.1? Why do you need an explicit package reference? I think if your app targets netcoreapp2.1, you should get the API.

If you want to consume packages/libraries built from corefxlab, include the following MyGet feed: https://dotnet.myget.org/feed/dotnet-corefxlab/package/nuget/System.IO.FileSystem.Watcher.Polling

Let me know if you are still blocked. If you could share the csproj of your console application, I could take a look and help further.

Reply:
This is my CSPROJ:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

</Project>
```

I've also attached a ZIP of the (very tiny) project. [ConsoleApp1.zip](https://github.com/dotnet/corefxlab/files/1921680/ConsoleApp1.zip)

And notice that the Enumeration namespace is missing in my Object Explorer:
![image](https://user-images.githubusercontent.com/1784452/38899655-40b50638-425e-11e8-9235-2c8595eb70f2.png)

Reply:
```diff
- using System.IO.Enumerable; // namespace not found
+ using System.IO.Enumeration; // this is the right namespace
```

```diff
- FileSystemEnumerable test;  // type not found
+ FileSystemEnumerable<int> test;  // type is generic
```

Reply:
Thanks for the corrections, but the type/namespace still doesn't load.

Reply:
Next question: The `PollingWatcher` is currently targetting .NET Standard 1.3. Even if I target .NET Standard, it still won't include your new APIs. Can I make the PollingWatcher project target .NET Core? I don't know how concerned we are with compatibility with the .NET Framework.

Reply:
> Sigh. I accidentally installed Preview 1. Is there a way we can mark old releases [here](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300-preview1)?

cc @livarcocc 

> Can I make the PollingWatcher project target .NET Core? I don't know how concerned we are with compatibility with the .NET Framework.

I don't own this project so I can't make the call. If we need to take a dependency on APIs only available on a higher version of net standard, or .net core 2.1 only (and we are ok with making this library less portable), then updating the target version would be fine. I don't think any other projects in corefxlab depend on this one. However, it is usually recommended for portable libraries to target the lowest possible version (and maybe have if/defs, if absolutely necessary).

cc @JeremyKuhne

Reply:
Yes, go ahead and target 2.1. We don't want to rewrite the interop logic and until we find that we really have to have this down level we can address that later. It should be designed around the high-performance APIs I've mentioned. :)

@KrzysztofCwalina, if you have any concerns we can discuss, but I believe this is in line with what you were expecting.